app/models/user.rb

```diff
@@ -293,10 +293,34 @@ def teacher?
     given_lectures.any?
   end
 
+  def teachable_editor?
+    edited_courses.any? || edited_lectures.any?
+  end
+
+  def teachable_editor_or_teacher?
+    teachable_editor? || teacher?
+  end
+
+  def can_edit_teachables?
+    admin? || teachable_editor_or_teacher?
+  end
+
+  # if you are not a teacher of lecture or a module editor,
+  # but just an additional editor of some lecture, you
+  # will not be considered active if all of your edited lectures
+  # are older than one year
+  def active_teachable_editor?
+    return false unless can_edit_teachables?
+    return true if admin || course_editor? || teacher?
+
+    edited_lectures.select { |l| l.term.nil? || !l.older_than?(1.year)}
+                   .any?
+  end
```

Comment:
I don't get the wording `active_teachable_editor` here. "Teachable" means "able to learn" (lernfähig) and I don't understand what it means in this context. It rather sounds like it tries to describe the students themselves; them being hopefully "teachable".

You define a teachable editor to be a user that has edited any course _or_ has edited any lecture (see line 297). For me, that does not sound related to a teacher, but rather just like an "editor" of a lecture or a course (Veranstaltungseditor:in).

According to line 324, the only difference between a "teachable_editor" and a normal "editor" is that an editor might be someone that has edited any media other than "Talks". So what is the difference then really? How can someone edit media without having editor rights (= without being a "Zusätzlicher Veranstaltungseditor:in")?

Reply:
"Teachable" is an artificial word that we created to describe things that can be taught. I will use it in the following explanation. Usually, polymorphic relations in rails are ending with -able see e.g. [here](https://guides.rubyonrails.org/association_basics.html#polymorphic-associations). A teachable in general can be a course/lecture/lesson/talk. Each medium belongs to a teachable via a polymorphic relation. Among these teachables, only courses/lectures/talks have associated editors (in a relation sense). Lessons can be edited by all editors of the corresponing lecture (and users with higher rights, like the course editor of the course the lecture is in). Also, each medium has an editor. A `teachable_editor` in the sense of line 297 can be a course editor (=Modulverantwortlicher) or a lecture editor. An `editor` in the sense of line 322 is a `teachable_editor` or an editor of a medium that is not a talk. We exclude talk editors (Vortragseditoren) for seminars talks here because these are typically students in seminars that are not supposed to have elevated rights: they are supposed to only edit their talk and create media for it. It is possible that someone can edit media without being a course or lecture editor. E.g. when you were such a course/lecture editor at some point in time and created your medium then: you wil keep the editing rights for this medium (but not the lecture) even if you lose the editing rights for the course/lecture. Also, you could be a student who created a medium for a talk in a seminar. This is indeed all a bit messy.